src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -20,30 +24,56 @@ import {
   MOBILE_AND_TABLET_CONTENT_HEIGHT,
   palette,
 } from '@/components/constants';
-
 import Header from './Header';
 import MessageField from './MessageField';
 import MessageList from './MessageList';
 
 const ActiveWindow = () => {
   const { isTablet } = useGetLayoutMode();
   const dispatch = useAppDispatch();
+  const [sendMessage, { isLoading: isLoadingSendMessage }] =
+    useSendMessageMutation();
+  const userId = useAppSelector(selectUserId);
 
   const activeChat = useAppSelector(selectActiveChat);
   const defaultChat = useAppSelector(selectDefaultChat);
   const chat = activeChat ?? defaultChat;
 
-  useEffect(() => {
-    dispatch(setActiveChat(chat.buddyId));
-  }, [chat.buddyId]);
-
-  const [, { isLoading: isLoadingSendMessage }] = useSendMessageMutation();
+  const [message, setMessage] = useState('');
 
   const isLoadingMessages = useAppSelector(
     selectIsLoadingBuddyMessages(chat?.buddyId),
   );
+  const isLoading = isLoadingMessages || isLoadingSendMessage;
+
+  const [isLoadingNewMessage, setIsLoadingNewMessage] = useState(false);
+  const isSendingDisabled = isLoadingNewMessage || message.trim().length < 1;
 
-  const isLoading = isLoadingSendMessage || isLoadingMessages;
+  const handleMessageSend = async () => {
+    if (!userId || isLoadingNewMessage) return;
+
+    setIsLoadingNewMessage(true);
+    try {
+      await sendMessage({
+        userId,
+        message: toSendMessage(chat.buddyId, userId, message),
+      }).unwrap();
+    } catch (error) {
+      setIsLoadingNewMessage(false);
+    }
+  };
+
+  useEffect(() => {
+    dispatch(setActiveChat(chat.buddyId));
```

Comment:
Also, what happens if this is removed?
I may not understand this either.
In my head we change the activechat when we click on the names on the leftside buddylist. Is there some apparent reason that I dont see, why this is done?

Reply:
I agree that it's not the most understandable solution.

![Näyttökuva 2024-11-14 kello 16 14 07](https://github.com/user-attachments/assets/4bd42242-373d-4b55-bc26-ca75f562a928)

If there's no selected active chat when this component is mounted, we get the default chat and then set it as the active chat.

Reply:
Good point! Fixed! 💪